recipes/muparserx/all/conanfile.py

```diff
@@ -0,0 +1,49 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class MuparserxConan(ConanFile):
+    name = "muparserx"
+    description = "Math Parser Library"
+    license = "BSD"
+    topics = ("conan", "muparserx", "math", "parser")
+    homepage = "https://beltoforion.de/article.php?a=muparserx"
+    url = "https://github.com/beltoforion/muparserx"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": False}
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.configure(source_folder=self._source_subfolder)
```

Comment:
This is wrong. You must:
- add the `cmake` generator;
- create a `CMakeLsts.txt` that will do at least `include(conanbuildinfo.cmake)`, `conan_basic_setup()` and `add_subdirectory(source_subfolder)`;
- add it to `exports_sources`; and
- Remove the `source_subfolder=self._source_subfolder` argument from the configure

Reply:
What does it mean "add the cmake generator"? The remaining points are clear to me now that you pointed them out.

Reply:
Add `generators = "cmake"` to the recipe such that conan will create a `conanbuildinfo.cmake` file in the build folder.